x-pack/plugin/searchable-snapshots/src/test/java/org/elasticsearch/index/store/cache/CacheFileTests.java

```diff
@@ -367,6 +366,31 @@ public void onEviction(CacheFile evictedCacheFile) {
         }
     }
 
+    private static class TestCacheFileModificationListener implements CacheFile.ModificationListener {
+
+        private final Set<CacheFile> updates = new HashSet<>();
+        private final Set<CacheFile> deletes = new HashSet<>();
+
+        @Override
+        public synchronized void onCacheFileUpdate(CacheFile cacheFile) {
+            updates.add(cacheFile);
```

Comment:
I think we can since `CacheFile` has a built in `needsFsync` to prevent more than one `onCacheFileNeedsFsync` notification at a time. I pushed c5759896f437182f21f4c243aadb3132028537c2.